packages/statemanager/src/statefulVerkleStateManager.ts

```diff
@@ -0,0 +1,344 @@
+        const basicDataBytes = encodeVerkleLeafBasicData(account)
+        await this._trie.put(
+          stem,
+          [VerkleLeafType.BasicData, VerkleLeafType.CodeHash],
+          [basicDataBytes, account.codeHash],
+        )
+      } else {
+        // Delete account
+      }
+    } else {
+      if (account !== undefined) {
+        this._caches?.account?.put(address, account, true)
+      } else {
+        this._caches?.account?.del(address)
+      }
+    }
+  }
+
+  deleteAccount = async (address: Address): Promise<void> => {
+    if (this.DEBUG) {
+      this._debug(`Delete account ${address}`)
+    }
+
+    this._caches?.deleteAccount(address)
+
+    if (this._caches?.account === undefined) {
+      const stem = getVerkleStem(this.verkleCrypto, address)
+      // TODO: Determine the best way to clear code/storage for an account when deleting
+      // Will need to inspect all possible code and storage keys to see if it's anything
+      // other than untouched leaf values
+      await this._trie.del(stem, [VerkleLeafType.BasicData, VerkleLeafType.CodeHash])
+    }
+  }
+
+  modifyAccountFields = async (address: Address, accountFields: AccountFields): Promise<void> => {
+    await modifyAccountFields(this, address, accountFields)
+  }
+  putCode = async (address: Address, value: Uint8Array): Promise<void> => {
+    if (this.DEBUG) {
+      this._debug(`putCode address=${address.toString()} value=${short(value)}`)
+    }
+
+    this._caches?.code?.put(address, value)
+
+    const codeHash = keccak256(value)
+    if (KECCAK256_NULL === codeHash) {
+      // If the code hash is the null hash, no code has to be stored
+      return
+    }
+
+    if ((await this.getAccount(address)) === undefined) {
+      await this.putAccount(address, new Account())
+    }
+    if (this.DEBUG) {
+      this._debug(`Update codeHash (-> ${short(codeHash)}) for account ${address}`)
+    }
+
+    const codeChunks = chunkifyCode(value)
+    const chunkStems = await generateCodeStems(codeChunks.length, address, this.verkleCrypto)
+
+    const chunkSuffixes: number[] = generateChunkSuffixes(codeChunks.length)
+    // Put the code chunks corresponding to the first stem (up to 128 chunks)
+    await this._trie.put(
+      chunkStems[0],
+      chunkSuffixes.slice(0, codeChunks.length <= 128 ? codeChunks.length : 128),
+      codeChunks.slice(0, codeChunks.length <= 128 ? codeChunks.length : 128),
+    )
+    // Put additional chunks under additional stems as applicable
+    for (let stem = 1; stem < chunkStems.length; stem++) {
+      await this._trie.put(
+        chunkStems[stem],
+        chunkSuffixes.slice(
+          128 + (256 * stem - 1),
+          codeChunks.length <= 128 + 256 * stem ? codeChunks.length : 128 + 256 * stem,
+        ),
+        codeChunks.slice(
+          128 + (256 * stem - 1),
+          codeChunks.length <= 128 + 256 * stem ? codeChunks.length : 128 + 256 * stem,
+        ),
```

Comment:
Where are you seeing this?  In the code below here, it's not `stem - 1` but `(VERKLE_NODE_WIDTH * stem) - 1` since these arrays are zero-indexed.  I'm not referencing a specific chunkId but a section of the overall array of chunks corresponding to the stem we're putting them under.